gorb/gorb.go

```diff
@@ -0,0 +1,323 @@
+	default:
+		return fmt.Errorf("unsupported loopback action %s", action)
+	}
+
+	var errorArr *multierror.Error
+	vipLoadbalancerArr := strings.Split(g.vipLoadbalancer, ",")
+	for index, vip := range vipLoadbalancerArr {
+		log.Infof(fmt.Sprintf("VIP %s loopback: %s", action, vip))
+		cmd := fmt.Sprintf("ip addr %s %s/32 dev lo label lo:%d", action, vip, index)
+		_, errCmd := exec.Command("bash", "-c", cmd).Output()
+		errorArr = multierror.Append(errorArr, errCmd)
+
+		log.WithFields(log.Fields{"arp_ignore": arpIgnore}).Info("Set arp_ignore to: ")
+		cmd = fmt.Sprintf("echo %d > %s", arpIgnore, path.Join(g.interfaceProcFsPath, "arp_ignore"))
+		_, errCmd = exec.Command("bash", "-c", cmd).Output()
+		errorArr = multierror.Append(errorArr, errCmd)
+
+		log.WithFields(log.Fields{"arp_announce": arpAnnounce}).Info("Set arp_announce to: ")
+		cmd = fmt.Sprintf("echo %d > %s", arpAnnounce, path.Join(g.interfaceProcFsPath, "arp_announce"))
+		_, errCmd = exec.Command("bash", "-c", cmd).Output()
+		errorArr = multierror.Append(errorArr, errCmd)
+	}
+
+	return errorArr.ErrorOrNil()
+}
+
+// Stop removes this instance from Gorb
+func (g *gorb) Stop() error {
+	var errorArr *multierror.Error
+	for _, backend := range g.backend {
+		backend.BackendConfig.Weight = 0
+		err := g.modifyBackend(&backend)
+		if err != nil {
+			log.WithFields(log.Fields{"err": err}).Error("Unable to set the backend weight to 0")
+			errorArr = multierror.Append(errorArr, err)
+		} else {
+			log.WithFields(log.Fields{"backend": backend}).Infof("Set backend weight to 0")
+		}
+	}
+
+	log.Infof("Waiting %v to finish gorb draining", g.drainDelay)
+	time.Sleep(g.drainDelay)
+
+	if g.manageLoopback {
+		err := g.manageLoopBack(deleteLoopback)
+		errorArr = multierror.Append(errorArr, err)
+	}
+
+	for _, backend := range g.backend {
+		err := g.removeBackend(&backend)
+		if err != nil {
+			log.WithFields(log.Fields{"err": err}).Error("Unable to remove Backend ")
+			errorArr = multierror.Append(errorArr, err)
+		} else {
+			log.WithFields(log.Fields{"backend": backend}).Infof("Backend succesfully removed")
+		}
+	}
+
+	return errorArr.ErrorOrNil()
+}
+
+func (g *gorb) Health() error {
+	resp, err := g.httpClient.Get(fmt.Sprintf("%s/service", g.serverBaseURL))
+	if err != nil {
+		return fmt.Errorf("unable to check service details: %v", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("gorb server is not healthy. Status code: %d, Response Body %s", resp.StatusCode, resp.Body)
+	}
+
+	return nil
+}
+
+func (g *gorb) Update(controller.IngressEntries) error {
+	var errorArr *multierror.Error
+	if g.manageLoopback {
+		err := g.manageLoopBack(addLoopback)
+		errorArr = multierror.Append(errorArr, err)
+	}
+
+	for _, backend := range g.backend {
+
+		backendExists, err := g.backendExists(&backend)
+		if err != nil {
+			log.WithFields(log.Fields{"err": err, "backend": backend}).Error("Unable to check if backend already exists")
+		}
+
+		if backendExists == 404 {
```

Comment:
My prior comment still stands - this should be a bool, rather than asserting on status code directly.

Reply:
Also use `http.StatusNotFound`